iothub/device/src/Transport/AmqpIot/AmqpIotCbsTokenProvider.cs

```diff
@@ -57,5 +58,29 @@ public async Task<CbsToken> GetTokenAsync(Uri namespaceAddress, string appliesTo
                 }
             }
         }
+
+        public void Dispose()
+        {
+            // Do not change this code. Put cleanup code in 'Dispose(bool disposing)' method
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_isDisposed)
+            {
+                if (disposing)
+                {
+                    if (_connectionString?.TokenRefresher != null
+                        && _connectionString.TokenRefresher.InstanceCreatedBySdk)
+                    {
+                        _connectionString.TokenRefresher.Dispose();
```

Comment:
_connectionString.TokenRefresher = null after diposing as a best practice?

Reply:
I agree, but in this case the _connectionString is readonly and hence, set up only in the constructor. I didn't want to modify it, so as to protect against accidental reinitialization in the future.